### PR TITLE
Add tests for errorHelper.ts, fix messages for warnings, fix gulp task build asynchronicity

### PR DESCRIPTION
**Pull request by SounD120**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -107,10 +107,10 @@ function build(failOnError, buildNls) {
         .pipe(gulp.dest(function (file) {
             return file.cwd;
         }))
-        .once("error", () => {
+        .once("error", function () {
```
Comment (ruslan-bikkinin) [inline gulpfile.js]: What is the point of changing this to es5 syntax?

File: gulpfile.js [hunk 2]
```diff
@@ -107,10 +107,10 @@ function build(failOnError, buildNls) {
...
             gotError = true;
         })
-        .once("finish", () => {
```
Comment (ruslan-bikkinin) [inline gulpfile.js]: What is the point of changing this to es5 syntax?

File: gulpfile.js [hunk 3]
```diff
@@ -107,10 +107,10 @@ function build(failOnError, buildNls) {
...
+        .once("finish", function ()  {
```
Comment (ruslan-bikkinin) [inline gulpfile.js]: What is the point of changing this to es5 syntax?

File: gulpfile.js (hunk 1 above)
Comment (SounD120) [reply gulpfile.js]: Thanks for pointing. I have fixed that.

File: gulpfile.js (hunk 3 above)
Comment (SounD120) [reply gulpfile.js]: Thanks for pointing. I have fixed that.

File: gulpfile.js (hunk 2 above)
Comment (SounD120) [reply gulpfile.js]: Thanks for pointing. I had fixed that.

**Pull request merged**